packages/xo-server/src/api/vif.js

```diff
@@ -119,7 +119,7 @@ export async function set({
       mac,
       currently_attached: attached,
       ipv4_allowed: newIpAddresses,
-      locking_mode: lockingMode,
+      locking_mode: lockingMode ?? vif.lockingMode,
```

Comment:
On changing the network, keeping the IPs will not make sens regarding this [PR](#5358). So, i think keeping also the vif locking mode can be not appropriated because we are not in the same network issues.

Reply:
You link this PR in your comment.
The user can update the VIF's locking mode that's why we don't need that XO change it automatically.

Reply:
I talked about this PR https://github.com/vatesfr/xen-orchestra/pull/5358. The question is do we need to keep the VIF locking mode behavior on changing the network?

Reply:
@badrAZ Keeping the old locking mode value when changing the network is probably the expected behaviour in many cases. And if we don't want that, I'm not sure what value we could set it to.

Reply:
I thought about letting the default network behavior because maybe the new network doesn't have the same issues as the oldest. But ok with keeping the VIF locking mode.

Reply:
What kind of issues are you talking about?

The problem I see with automatically setting the locking mode to `network_default` is that if an admin purposely set the locking mode to `disabled` or `locked`, then the VM user can bypass the restriction by changing the network twice.